ofrak_ai/ofrak_ai/exponential_backoff.py

```diff
@@ -0,0 +1,77 @@
+"""
+MIT License
+
+Copyright (c) 2023 OpenAI
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import asyncio
+import random
+
+from openai.error import RateLimitError, OpenAIError
+
+# Modified from https://github.com/openai/openai-cookbook/blob/main/examples/How_to_handle_rate_limits.ipynb
+def retry_with_exponential_backoff(
+    func,
+    initial_delay: float = 1,
+    exponential_base: float = 2,
+    jitter: bool = True,
+    max_retries: int = 10,
+    errors: tuple = (RateLimitError,),
+):
+    """
+    Retry a function with exponential backoff.
+
+    AuthenticationError is NOT raised when the OPENAI_ORGANIZATION value is incorrect or missing.
+    Instead, the OpenAI API treats it as a RateLimitError.
+
+    :raises OpenAIError: if unable to make a valid request or receive a response from ChatGPT
+    """
+
+    async def wrapper(*args, **kwargs):
```

Comment:
```suggestion
    @functools.wraps
    async def wrapper(*args, **kwargs):
```
(You'll need to add the import as well)
This is generally a good thing to throw on any function decorator/wrapper: https://stackoverflow.com/questions/308999/what-does-functools-wraps-do